sequencers/modeshape-sequencer-audio/src/main/java/org/modeshape/sequencer/audio/AudioMetadataSequencer.java

```diff
@@ -0,0 +1,155 @@
+        setPropertyIfMetadataPresent(sequencedNode, AudioMetadataLexicon.SAMPLE_RATE, metadata.getSampleRate());
+        setPropertyIfMetadataPresent(sequencedNode, AudioMetadataLexicon.CHANNELS, metadata.getChannels());
+        setPropertyIfMetadataPresent(sequencedNode, AudioMetadataLexicon.LENGTH, metadata.getLength());
+
+        addTagNode(sequencedNode, metadata);
+        return true;
+    }
+
+    private void addTagNode( Node sequencedNode,
+                             AudioMetadata metadata ) throws ItemExistsException, PathNotFoundException, NoSuchNodeTypeException, LockException, VersionException, ConstraintViolationException, RepositoryException {
+        Node tagNode = sequencedNode.addNode(AudioMetadataLexicon.TAG_NODE, AudioMetadataLexicon.TAG_NODE);
+
+        setPropertyIfMetadataPresent(tagNode, AudioMetadataLexicon.TITLE, metadata.getTitle());
+        setPropertyIfMetadataPresent(tagNode, AudioMetadataLexicon.ARTIST, metadata.getArtist());
+        setPropertyIfMetadataPresent(tagNode, AudioMetadataLexicon.ALBUM, metadata.getAlbum());
+        setPropertyIfMetadataPresent(tagNode, AudioMetadataLexicon.YEAR, metadata.getYear());
+        setPropertyIfMetadataPresent(tagNode, AudioMetadataLexicon.COMMENT, metadata.getComment());
+        setPropertyIfMetadataPresent(tagNode, AudioMetadataLexicon.TRACK, metadata.getTrack());
+        setPropertyIfMetadataPresent(tagNode, AudioMetadataLexicon.GENRE, metadata.getGenre());
+
+        for (AudioMetadataArtwork artwork : metadata.getArtwork()) {
+            Node artworkNode = tagNode.addNode(AudioMetadataLexicon.ARTWORK_NODE, AudioMetadataLexicon.ARTWORK_NODE);
+            setPropertyIfMetadataPresent(artworkNode, JcrConstants.JCR_MIME_TYPE, artwork.getMimeType());
+            setPropertyIfMetadataPresent(artworkNode, AudioMetadataLexicon.ARTWORK_TYPE, artwork.getType());
+            setPropertyIfMetadataPresent(artworkNode, JcrConstants.JCR_DATA, artwork.getData());
+        }
+    }
+
+    private void setPropertyIfMetadataPresent( Node node,
+                                               String propertyName,
+                                               Object value) throws RepositoryException {
+        if (value != null) {
+            if (value instanceof String && !StringUtil.isBlank((String) value)) {
+                node.setProperty(propertyName, (String) value);
+            }
+            if (value instanceof Long) {
+                node.setProperty(propertyName, (Long) value);
+            }
+            if (value instanceof Integer) {
+                node.setProperty(propertyName, new Long((Integer) value));
+            }
+            if (value instanceof byte[]) {
+                InputStream is = new ByteArrayInputStream((byte []) value);
+                Binary binaryProperty = (Binary) node.getSession().getValueFactory().createBinary(is);
+                node.setProperty(propertyName, binaryProperty);
+            }
```

Comment:
these chain of `ifs` should be changed to `if elses` so that if `value` doesn't have any of the expected types, an exception (e.g. `IllegalArgumentException`) is thrown. 

This ensures that 
a) no value types are accidentally missed and 
b) if this is extended at some point in the future, this exception will flag the place where work needs to be done to support new value types